src/System.Device.Gpio/System/Device/Gpio/LibgpiodDriverEventHandler.cs

```diff
@@ -13,24 +13,26 @@ internal sealed class LibGpiodDriverEventHandler : IDisposable
     {
         private const int ERROR_CODE_EINTR = 4; // Interrupted system call
 
-        private static string s_consumerName = Process.GetCurrentProcess().ProcessName;
+        private static readonly string s_consumerName = Process.GetCurrentProcess().ProcessName;
 
         public event PinChangeEventHandler? ValueRising;
         public event PinChangeEventHandler? ValueFalling;
 
-        private int _pinNumber;
-        public CancellationTokenSource CancellationTokenSource;
-        private Task _task;
-        private bool _disposing = false;
+        private readonly int _pinNumber;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly Task _task;
+        private bool _disposing;
 
         public LibGpiodDriverEventHandler(int pinNumber, SafeLineHandle safeLineHandle)
         {
             _pinNumber = pinNumber;
-            CancellationTokenSource = new CancellationTokenSource();
+            _cancellationTokenSource = new CancellationTokenSource();
             SubscribeForEvent(safeLineHandle);
-            _task = InitializeEventDetectionTask(CancellationTokenSource.Token, safeLineHandle);
+            _task = InitializeEventDetectionTask(_cancellationTokenSource.Token, safeLineHandle);
         }
 
+        public CancellationToken CancellationToken => _cancellationTokenSource.Token;
```

Comment:
Similar to @pgrawehr question, what happens when somebody calls this property on a disposed Libgpiod event handler?

Reply:
Before this PR the whole `CancellationTokenSource` was pubic. Now it is the token only.